### PR TITLE
update resource-loading

### PR DESCRIPTION
**Pull request by kaycebasques**
@pbakaus please review this big ol' update of [Measure Resource Loading Times](https:/developers.google.com/web/tools/chrome-devtools/profile/network-performance/resource-loading).

@Garbee in one of our issues or pull requests (can't find the one) we talked about documenting all of the network filter keywords... I've documented them in this doc.

cc @paulirish

### CODE REVIEW

**Comment by kaycebasques** [comment]
@crhym3 does the build process automatically stage all PRs now? What's the URL for this one?

**Comment by Garbee** [comment]
Clicking on "Show all checks" and once the "builder" check has passed, the details link will take you to the build. In this case [https://pr-2585-dot-weasel-dot-web-central.appspot.com/](https://pr-2585-dot-weasel-dot-web-central.appspot.com/).

**Comment by Garbee** [comment]
A few quick things without having re-read the text content fully yet...

1) Should we continue with the previous [image guidelines](https://github.com/GoogleChrome/devtools-docs/blob/master/contributing-images.md)? Primarily color for hints. The black can be hard to tell sometimes in the images.
2) Per the old guidlines, images like the one showing the [DOMContentLoaded](https://pr-2585-dot-weasel-dot-web-central.appspot.com/web/tools/chrome-devtools/profile/network-performance/imgs/domcontentloaded.png) indicators should simply be arrows pointing to the areas. Maybe a small box around the line in the scrobbler since it can be hard to see. (The load image as well has this issue)
3) Quite a few images can be cropped to save space. For example [timeline-view-hover](https://github.com/kaycebasques/WebFundamentals/blob/resource-loading/src/content/en/tools/chrome-devtools/profile/network-performance/imgs/timeline-view-hover.png). This includes the entire DevTools UX which isn't necessary to get the example across. The previous image was just the hover area, so only the data needed to show what was being discussed was shown.
4) Images should flow after text content as much as possible. For example, the opening section starts with "The Network panel consists of five panes:" then gives an image and then the text flow continues. Instead the text flow should all be together and the image should be just before the next section begins. Text rules the content, images are supplementary extras.

I'll comment inline as I go over the content on textual issues. But these image problems are harder to handle inline.

File: src/content/en/tools/chrome-devtools/profile/network-performance/resource-loading.markdown [hunk 1]
```diff
@@ -138,133 +127,137 @@ Each requested resource is added as a row to the Network table, which contains t
...
 
-![Resource image preview](imgs/network-image-preview.png)
+The tabs available change depending on what type of resource you've selected,
+but the four tabs below are most common:
 
-And JSON resources:
+* **Headers**. View HTTP headers associated to the resource.
+* **Preview**. View previews of JSON, image, and text resources.
+* **Response**. View the resource's HTTP response data (if any).
+* **Timing**. View a granular breakdown of the request lifecycle for the
```
Comment (Garbee) [inline src/content/en/tools/chrome-devtools/profile/network-performance/resource-loading.markdown]: "View" seems repeated a lot here. Maybe just dump it and get right into the description?

File: src/content/en/tools/chrome-devtools/profile/network-performance/resource-loading.markdown [hunk 2]
```diff
@@ -138,133 +127,137 @@ Each requested resource is added as a row to the Network table, which contains t
...
 
-And JSON resources:
+* **Headers**. View HTTP headers associated to the resource.
+* **Preview**. View previews of JSON, image, and text resources.
+* **Response**. View the resource's HTTP response data (if any).
+* **Timing**. View a granular breakdown of the request lifecycle for the 
+  resource.
 
-![Resource JSON preview](imgs/resource-preview-json.png)
+### View network timing
```
Comment (Garbee) [inline src/content/en/tools/chrome-devtools/profile/network-performance/resource-loading.markdown]: Let's try to find a way to link over to the understanding resource timing doc from within this section.

File: src/content/en/tools/chrome-devtools/profile/network-performance/resource-loading.markdown [hunk 3]
```diff
@@ -138,133 +127,137 @@ Each requested resource is added as a row to the Network table, which contains t
...
+<!-- the screenshot above and list below are redundant, but we include
+     the text for SEO -->
 
-The Cookies tab displays a table of all the cookies transmitted in the
-resource's HTTP request and response headers. You can also clear all cookies.
+* Queuing
+* Stalled
+* Request/Response
+* Request sent
+* Waiting (Time to first byte (TTGB))
```
Comment (Garbee) [inline src/content/en/tools/chrome-devtools/profile/network-performance/resource-loading.markdown]: TTGB -> TTFB

File: src/content/en/tools/chrome-devtools/profile/network-performance/resource-loading.markdown [hunk 4]
```diff
@@ -385,122 +388,213 @@ The Frames tab shows messages sent or received over a WebSocket connection. This
...
+  with a `Domain` attribute that matches the specified value. DevTools 
+  populates the autocomplete with all of the cookie domains that it has 
+  encountered.
+* `set-cookie-name`: Show the resources that have a `Set-Cookie` header 
+  with a name that matches the specified value. DevTools populates the 
+  autocomplete with all of the cookie names that it has encountered.
+* `set-cookie-value`: Show the resources that have a `Set-Cookie` header
+  with a value that matches the specified value. DevTools populates the 
+  autocomplete with all of the cookie values that it has encountered.
+* `status-code`: Only show resources whose HTTP status code match the
```
Comment (Garbee) [inline src/content/en/tools/chrome-devtools/profile/network-performance/resource-loading.markdown]: This list is not styles like other lists. It uses code formatting vs bolding for the items being defined. Should probably update this list to by styled like the others for consistency. Also verifying each definition list is styled consistently.

File: src/content/en/tools/chrome-devtools/profile/network-performance/resource-loading.markdown [hunk 5]
```diff
@@ -385,122 +388,213 @@ The Frames tab shows messages sent or received over a WebSocket connection. This
...
+
+## Filter requests 
+
+The **Network** panel provides numerous ways to filter which resources are 
+displayed. Click the **filters** button
+(![filters button](imgs/filters.png){:.inline})
+to hide or display the **Filters** pane.
+
+Use the content type buttons to only display resources of the selected 
+content type. Tip: hold <kbd>cmd</kbd> and then click on one of these
```
Comment (Garbee) [inline src/content/en/tools/chrome-devtools/profile/network-performance/resource-loading.markdown]: The tip needs alt styling to call it out. Should probably use a [note](https://web-central.appspot.com/web/resources/jekyll/useful-and-common-tags?hl=en) type thing here. Also mention Cmd or Ctrl for Windows/Linux.

File: src/content/en/tools/chrome-devtools/profile/network-performance/resource-loading.markdown [hunk 6]
```diff
@@ -385,122 +388,213 @@ The Frames tab shows messages sent or received over a WebSocket connection. This
...
+Large rows enable some columns to display two text fields: a primary 
+field and a secondary field. The column header indicates the meaning of the 
+secondary field. 
 
 ### Add and remove table columns
 
-Change the default set of columns displayed by the Network table.
-To show or hide a column,
-<span class="kbd">Right-click</span> or <span class="kbd">Ctrl</span> + <span class="kbd">Click</span> (Mac only) in the table header and select or deselect column names from the list.
+Click on any of the headers in the **Requests Table** to add or remove
```
Comment (Garbee) [inline src/content/en/tools/chrome-devtools/profile/network-performance/resource-loading.markdown]: **Right** click. Just clicking sorts by that column.

File: src/content/en/tools/chrome-devtools/profile/network-performance/resource-loading.markdown [hunk 7]
```diff
@@ -385,122 +388,213 @@ The Frames tab shows messages sent or received over a WebSocket connection. This
...
+then select **Copy as cURL**, DevTools recreates an HTTP request (including
+HTTP headers, SSL certificates, and query string parameters) and copies it 
+as a cURL command string to the clipboard. You can then paste the string 
+into a terminal window (on a system with cURL) to execute the same request.
 
-Below is an example cURL command line string taken from a XHR request on the Google News home page.
+Below is an example cURL command line string taken from a XHR request on 
+the Google News home page.
 
 `curl 'http://news.google.com/news/xhrd=us' -H 'Accept-Encoding: gzip,deflate,:sdch' -H 'Host: news.google.com' -H 'Accept-Language: en-US,en;q=0.8' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/29.0.1510.0 Safari/537.36' -H 'Accept: */*' -H 'Referer: http://news.google.com/nwshp?hl=en&tab=wn' -H 'Cookie: NID=67=eruHSUtoIQA-HldQn7U7G5meGuvZOcY32ixQktdgU1qSz7StUDIjC_Knit2xEcWRa-e8CuvmADminmn6h2_IRpk9rWgWMdRj4np3-DM_ssgfeshItriiKsiEXJVfra4n; PREF=ID=a38f960566524d92:U=af866b8c07132db6:FF=0:TM=1369068317:LM=1369068321:S=vVkfXySFmOcAom1K' -H 'Connection: keep-alive' --compressed`
```
Comment (Garbee) [inline src/content/en/tools/chrome-devtools/profile/network-performance/resource-loading.markdown]: Maybe run this through the [bash highlighter](https://web-central.appspot.com/web/resources/jekyll/useful-and-common-tags?hl=en#highlight) to see if it helps the output look better at all.

**Comment by kaycebasques** [comment]
Right on @Garbee good feedback

**Comment by kaycebasques** [comment]
@Garbee I agree with all four of your feedback items regarding images

File: src/content/en/tools/chrome-devtools/profile/network-performance/resource-loading.markdown (hunk 4 above)
Comment (kaycebasques) [inline src/content/en/tools/chrome-devtools/profile/network-performance/resource-loading.markdown]: This is intentional. UI elements (e.g. buttons) are bold. Literal text is formatted as code.

**Comment by kaycebasques** [comment]
@Garbee I hit all your points, thanks for the review

**Comment by pbakaus** [comment]
This is great work! I just did a quick pass and it looks really good. I actually just stumbled upon the old version of this page today and was looking for an explanation to "Waiting (TTFB)", but didn't find any.

In this new updated guide, we still don't explain it. Can we add an explanation for at least Queuing and Waiting, as many people have trouble understanding these?

**Comment by Garbee** [comment]
> In this new updated guide, we still don't explain it. Can we add an explanation for at least Queuing and Waiting, as many people have trouble understanding these?

This is what the new "Understanding Resource Timing" doc is for, from PR #2516.

**Comment by Garbee** [comment]
LGTM. Thanks for updating the images so quickly!

**Pull request merged**